sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -567,13 +550,46 @@ private[sbtplugin] object ScalaJSPluginInternal {
         log.debug(s"with JSEnv ${env.name}")
 
         val input = jsEnvInput.value
-        val config = baseRunConfig.withLogger(sbtLogger2ToolsLogger(log))
+
+        /* The list of threads that are piping output to System.out and
+         * System.err. This is not an AtomicReference or any other thread-safe
+         * structure because the `onOutputStream` callback is always called
+         * within the thread that calls `env.start()`.
```

Comment:
Hum, nothing to my knowledge in `JSEnv`'s spec guarantees this.

That being said, I think it is OK if this is a `var`, but its a bit more complicated:
- `onOutputStream` is guaranteed to be called exactly once.
- `pipeOutputThreads` is only read once the run is completed (while the `JSEnv` interface does not explicitly specify that the call to `onOutputStream` must happen before that, anything else is just plain unreasonable).

This explanation is missing an argument that there is an actual (chain) of memory barriers between the write(s) to `pipeOutputThreads` and the subsequent read. But that's probably OK...